File: src/renders/facilities/park/park.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Park.cpp             */

#include <iostream>
#include "park.h"
using namespace std;

/***** PARK *****/
Park::Park() : Facility('*', WHITE)
{}

Park::Park(const Park& P) : Facility(P)
{}

Park::~Park()
{}

Park& Park::operator=(const Park& P)
{
	Facility::operator=(P);
  return *this;
}

char Park::render()
{
	return id;
}

Color Park::getColor()
{
	return color;
}

Park* Park::clone() const {
  return new Park(*this);
}

File: src/facility/park/Park.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Park.java
 */

package facility.park;

import java.util.*;

import facility.Facility;

import java.lang.*;

public class Park extends Facility{
   /**
   * Constructor
   */
  public Park(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return '*';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
